tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/descartes/impl/descartes_motion_planner.hpp

```diff
@@ -118,25 +118,66 @@ tesseract_common::StatusCode DescartesMotionPlanner<FloatType>::solve(const Plan
 
   // Loop over the flattened results and add them to response if the input was a plan instruction
   Eigen::Index dof = problem->manip_fwd_kin->numJoints();
   Eigen::Index result_index = 0;
-  for (auto& instruction : results_flattened)
+  bool first_plan_instruction_found = false;
+  for (std::size_t plan_index = 0; plan_index < results_flattened.size(); plan_index++)
   {
-    if (isPlanInstruction(instruction.get()))
+    if (isPlanInstruction(instructions_flattened.at(plan_index).get()))
     {
-      // This instruction corresponds to a composite. Set all results in that composite to the results
-      auto* move_instructions = instruction.get().cast<CompositeInstruction>();
-      for (auto& instruction : *move_instructions)
+      const auto* plan_instruction = instructions_flattened.at(plan_index).get().cast_const<PlanInstruction>();
+
+      if (plan_instruction->isLinear())
+      {
+        // This instruction corresponds to a composite. Set all results in that composite to the results
+        auto* move_instructions = results_flattened[plan_index].get().cast<CompositeInstruction>();
+        for (auto& instruction : *move_instructions)
+        {
+          Eigen::Map<const Eigen::Matrix<FloatType, Eigen::Dynamic, 1>> temp(solution.data() + dof * result_index++,
+                                                                             dof);
+          instruction.cast<MoveInstruction>()->setPosition(temp.template cast<double>());
+        }
+      }
+      else if (plan_instruction->isFreespace())
       {
-        Eigen::Map<Eigen::Matrix<FloatType, -1, 1>> temp(solution.data() + dof * result_index, dof);
-        instruction.cast<MoveInstruction>()->setPosition(temp.template cast<double>());
-        result_index++;
+        // Because descartes does not support freespace it just includes the plan instruction waypoint so we will
+        // fill out the results with a joint interpolated trajectory.
+        Eigen::Map<const Eigen::Matrix<FloatType, Eigen::Dynamic, 1>> start(solution.data() + dof * result_index, dof);
+        Eigen::Map<const Eigen::Matrix<FloatType, Eigen::Dynamic, 1>> stop(solution.data() + dof * result_index++, dof);
+
+        // This instruction corresponds to a composite. Set all results in that composite to the results
+        auto* move_instructions = results_flattened[plan_index].get().cast<CompositeInstruction>();
+
+        if (!first_plan_instruction_found)
+        {
+          Eigen::MatrixXd temp = interpolate(start.template cast<double>(),
+                                             stop.template cast<double>(),
+                                             static_cast<int>(move_instructions->size()) - 1);
+
+          assert(temp.cols() == static_cast<long>(move_instructions->size()));
+          for (std::size_t i = 0; i < move_instructions->size(); ++i)
+            (*move_instructions)[i].cast<MoveInstruction>()->setPosition(temp.col(static_cast<long>(i)));
```

Comment:
So this is still assuming we don't fill out the start waypoint in the response composites?